### PR TITLE
Enhance uefi test suites

### PR DESCRIPTION
**Pull request by waynechen55**
* **Add** SLE 15-SP5 uefi guests profiles:
  * sles_15_sp5_64_kvm_hvm_uefi_with_power_management_x86_64.xml
  * sles_15_sp5_64_xen_hvm_uefi_with_power_management_x86_64.xml
  * sles_15_sp5_64_kvm_hvm_uefi_without_power_management_x86_64_using_virtmanagerlt3.0.0.xml
  * sles_15_sp5_64_xen_hvm_uefi_without_power_management_x86_64_using_virtmanagerlt3.0.0.xml
    
* **Update** SLE 15-SP4 guests profiles:
  * sles_15_sp4_64_kvm_hvm_uefi_with_power_management_x86_64.xml
  * sles_15_sp4_64_xen_hvm_uefi_with_power_management_x86_64.xml
    
* **Use** timeout command with options --kill-after and --signal to make sure command returns after timeout instead of hang in the system sometimes.

* **Verification runs:**
  * [sev-es test](https://openqa.suse.de/tests/9707066) Failed due to bsc#1203923
  * [15sp5 on 15sp5 uefi kvm](https://openqa.suse.de/tests/9707350) Failed due to bsc#1204220
  * [15sp5 on 15sp5 uefi xen](https://openqa.suse.de/tests/9752148)
  * [15sp5 on 15sp4 uefi kvm](https://openqa.suse.de/tests/9756424)
  * [15sp5 on 15sp4 uefi xen](https://openqa.suse.de/tests/9752186)
  * [15sp5 on 12sp5 uefi kvm](https://openqa.suse.de/tests/9752354)
  * [15sp5 on 12sp5 uefi xen](https://openqa.suse.de/tests/9752184)
  * [15sp4 on 15sp5 uefi kvm](https://openqa.suse.de/tests/9707981) Failed due to bsc#1204220
  * [15sp4 on 15sp5 uefi xen](https://openqa.suse.de/tests/9707982)
  * [12sp5 on 15sp5 uefi kvm](https://openqa.suse.de/tests/9707933) Failed due to bsc#1204220
  * [12sp5 on 15sp5 uefi xen](https://openqa.suse.de/tests/9707374)
  * [oraclelinux on 15sp5](https://openqa.suse.de/tests/9714856)
  * [sle micro](https://openqa.suse.de/tests/9715288)

### CODE REVIEW

**Comment by waynechen55** [comment]
@alice-suse @Julie-CAO @guoxuguang @nanzhg @RoyCai7 Please have a look.

**Comment by Julie-CAO** [comment]
LGTM.

**Comment by guoxuguang** [comment]
LGTM.

File: data/virt_autotest/guest_params_xml_files/sles_15_sp5_64_xen_hvm_uefi_with_power_management_x86_64.xml [hunk 1]
```diff
@@ -84,7 +84,7 @@
     <guest_autoconsole>text</guest_autoconsole>
     <guest_noreboot></guest_noreboot>
     <guest_default_target>graphical</guest_default_target>
-    <guest_do_registration>true</guest_do_registration>
+    <guest_do_registration>false</guest_do_registration>
```
Comment (alice-suse) [inline data/virt_autotest/guest_params_xml_files/sles_15_sp5_64_xen_hvm_uefi_with_power_management_x86_64.xml:87]: I remember the registration bug is fixed, right? Can we switch on this?

File: lib/guest_installation_and_configuration_base.pm [hunk 2]
```diff
@@ -394,10 +394,10 @@ sub prepare_non_transactional_environment {
     if (!is_transactional) {
         virt_autotest::utils::setup_rsyslog_host($common_log_folder);
         my $_packages_to_check = 'wget curl screen dnsmasq xmlstarlet yast2-schema python3 nmap';
-        zypper_call("install -y $_packages_to_check");
+        zypper_call("install -y $_packages_to_check", exitcode => [0, 106]);
         my $_patterns_to_check = 'kvm_server kvm_tools';
         $_patterns_to_check = 'xen_server xen_tools' if ($self->{host_virt_type} eq 'xen');
-        zypper_call("install -y -t pattern $_patterns_to_check");
+        zypper_call("install -y -t pattern $_patterns_to_check", exitcode => [0, 106]);
```
Comment (alice-suse) [inline lib/guest_installation_and_configuration_base.pm:400]: Would you please share what else repositories failed to be refreshed in addition to virtupdate? In this PR, the allowed zypper errors mainly come from repository side. If the repositories do not provide packages that we'd like to test, eg kvm,xen,libvirt, allowing such errors can be fine, eg above line 397 change. But Virtupdate repo that we set is the official product update repo which provides updated versions of kvm,xen,libvirt. I am wondering if such repo has problem, shall we stop the test? Lower versions of packages than update repo is not our test target.

File: data/virt_autotest/guest_params_xml_files/sles_15_sp5_64_xen_hvm_uefi_with_power_management_x86_64.xml (hunk 1 above)
Comment (waynechen55) [reply data/virt_autotest/guest_params_xml_files/sles_15_sp5_64_xen_hvm_uefi_with_power_management_x86_64.xml]: Done. Verification updated.

File: lib/guest_installation_and_configuration_base.pm (hunk 2 above)
Comment (waynechen55) [reply lib/guest_installation_and_configuration_base.pm]: autobuild guys fixed the issue. Removed "exitcode". Done. Verification updated.

**Review by alice-suse — APPROVED** [review]

**Pull request merged**